app/services/submit_reference.rb

```diff
@@ -18,11 +18,41 @@ def save!
     if @send_emails
       CandidateMailer.reference_received(reference).deliver_later
       RefereeMailer.reference_confirmation_email(application_form, reference).deliver_later
+      notify_provider_users
     end
   end
 
 private
 
+  def notify_provider_users
+    return unless FeatureFlag.active?(:new_references_flow_providers)
+
+    application_choices.each do |ac|
+      ac.provider.provider_users.each do |pu|
+        ProviderMailer.reference_received(
+          provider_user: pu,
+          application_choice: ac,
+          reference: reference,
+          course: course,
+          ordinance: ordinance,
+        ).deliver_later
+      end
+    end
+  end
+
+  # i.e 1st, 2nd, 3rd
+  def ordinance
+    @ordinance ||= TextOrdinalizer.new(reference.order_in_application_references).call
+  end
+
+  def course
+    @course ||= reference.candidate.course_from_find
+  end
+
+  def application_choices
+    @application_choices ||= reference.application_form.application_choices.pending_conditions
+  end
```

Comment:
Minor: could rename method to make it clearer that it's only returning the choice which the candidate accepted an offer for (which should only ever be a single choice)?

Also think it should include 'recruited' as well as 'pending_conditions' states and possibly 'deferred'?

```suggestion
  def application_choice_accepted
    @application_choices ||= reference.application_form.application_choices.where(status: [:pending_conditions, :recruited])
  end
```

Reply:
Cool, we are aligned! I already changed this locally yesterday to `accepted_application`. I still didn't push all the fixes/changes because it is broken until this PR is merged: https://github.com/DFE-Digital/apply-for-teacher-training/pull/7449